### DISCUSSION TITLE
Curriculum learning possible?

### DISCUSSION THREAD

**Discussion by FabPi** [lululxvi/deepxde#734] [Q&A] [answered]
Hello,

i have tested a few things with DeepXDE and am very excited about the many possibilities! However, I have one question: Is it possible to do some kind of curriculum learning with the library?

I have tried the following, unfortunately without success:
Training the PINN with simplified boundary conditions and favorable values of the variables in the PDE. Then save the model. As soon as I change something in the variables in the PDE, I get errors when restoring the saved model. For the same configuration as before, it works as a charm.

Any chance to transfer the weights and biases but train with a new PDE or/and new boundaries?

Thanks for all your work.
Best regards

Fabian

Error Message for _model.restore()_ with changed variables:
_NotFoundError: 2 root error(s) found.
  (0) Not found: Key beta1_power not found in checkpoint
	 [[node save_1/RestoreV2 (defined at ProgramData\Anaconda3\envs\nn\lib\site-packages\deepxde\model.py:128) ]]
  (1) Not found: Key beta1_power not found in checkpoint
	 [[node save_1/RestoreV2 (defined at ProgramData\Anaconda3\envs\nn\lib\site-packages\deepxde\model.py:128) ]]
	 [[save_1/RestoreV2/_43]]
0 successful operations.
0 derived errors ignored._

**Comment by lululxvi** [comment, marked as answer]
See https://github.com/mariusmerkle/TL-PINNs

**Reply by FabPi** [reply]
Thank you for the answer. With the help of the code from the Thesis I was able to solve the problem.

Should anyone encounter a similar problem:
If you define a variable outside the PDE but use it inside the PDE, it causes said error. Therefore either an enclosing function must be created, which gets the value of the variable passed (see code in the Thesis) or the number must be entered directly in the PDE.

So either:

```
def training(var):
    def pde(x, u):
        ... + var + ...

if __name__ == "__main__":
    var = 5
    training(var)
```

or directly in the PDE:

```
def pde(x, u):
    ... + 5 + ...
```
